GYBITG/GalleryViewController.swift

```diff
@@ -143,11 +150,50 @@ class GalleryViewController: UITableViewController, UINavigationControllerDelega
             let dataPath = documentsDirectory.appendingPathComponent(videoFileName)
             try! videoData?.write(to: dataPath, options: [])
             }
+            //if the selected video is from the photoLibrary
+            else if( picker.sourceType == UIImagePickerController.SourceType.photoLibrary){
+            //check if the selected video is over the limit
+                //turn the selected video into an asset to get the duration
+                let asset = AVURLAsset(url: selectedVideo, options: nil)
+                
+                if( asset.duration.seconds > 180.0){
```

Comment:
I recommend you make the duration limit a variable so that it can be easily changed by developers